src/compiler/core.ts

```diff
@@ -942,15 +1030,26 @@ namespace ts {
      * Adds the value to an array of values associated with the key, and returns the array.
      * Creates the array if it does not already exist.
      */
-    export function multiMapAdd<V>(map: Map<V[]>, key: string | number, value: V): V[] {
-        const values = map[key];
+    export function multiMapAdd<V>(map: Map<V[]>, key: string, value: V): V[] {

```

Comment:
Maybe we should consider adding a `createMultiMap<T>` that returns a `MultiMap<T>`:

```ts
interface MultiMap<T> extends Map<T[]> {
  add(key: string, value: T): T[];
  remove(key: string, value: T): void;
}
function createMultiMap<T>() {
  const map = <MultiMap<T>>createMap<T[]>();
  map.add = multiMapAdd;
  map.remove = multiMapRemove;
  return map;
}
function multiMapAdd<T>(this: MultiMap<T>, key: string, value: T) { ... }
function multiMapRemove<T>(this: MultiMap<T>, key: string, value: T) { ... }